### PR TITLE
refactor(streaming): remove dist key for global simple agg & rename to `GlobalSimpleAgg`

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As title. We reuse `SimpleAggNode` for both global and local ones, where `dist_key` should be unused for global simple agg.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: proto/stream_plan.proto [hunk 1]
```diff
@@ -56,6 +56,7 @@ message MaterializeNode {
 // Local and global aggregator distinguish with each other in PlanNode definition.
 message SimpleAggNode {
   repeated expr.AggCall agg_calls = 1;
+  // Not used for global simple agg.
   repeated uint32 distribution_keys = 2;
```
Comment (fuyufjh) [inline proto/stream_plan.proto:60]: why not deleted?

Comment (BugenZhao) [reply proto/stream_plan.proto:60]: `SimpleAggNode` is also used for local agg. 🤣

**Review by fuyufjh — APPROVED** [review]

**Review by kwannoel — APPROVED** [review]
LGTM, thanks for doing this refactor :) Missed this in earlier PR.

**Review by BowenXiao1999 — APPROVED** [review]
LSTM

BTW can we also change the name from "StreamSimpleAgg" to "StreamGlobalSimpleAgg" in this PR? 😢

**Review by xx01cyx — APPROVED** [review]

**Comment by BugenZhao** [comment]
> StreamSimpleAgg

Cool.

**Pull request merged**